### ISSUE TITLE
Impossible to catch RPC errors via try...catch block

### ISSUE THREAD

**Issue by NyaomiDEV** [xhayper/discord-rpc#48]
**Describe the bug**
RPC errors don't bubble up, so us developers cannot give context to users and we are forced to fail.

**To Reproduce**
Try to connect to Discord RPC while not having Discord open using the `simple_status` example code snippet. An `RPC_COULD_NOT_CONNECT` error will be generated. Try to catch that error and handle it gracefully.

**Expected behavior**
Errors should bubble up to where us developers can handle them.

**(please complete the following information):**
 - OS: Arch Linux
 - OS Version: rolling
 - Discord Version: Not applicable

**Additional context**
```
(node:24271) UnhandledPromiseRejectionWarning: RPC_COULD_NOT_CONNECT: Could not connect
    at /**CENSORED**/node_modules/@xhayper/discord-rpc/dist/transport/IPC.js:113:20
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:24271) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
```

**Comment by xhayper** [comment]
Hello!

I don't quite understand the term "bubble up", is it possible for you to explain the term for me? and perhaps provide some examples?

**Comment by NyaomiDEV** [comment]
The idea is that if I have this snippet of code running while Discord is closed:
```js
async () => {
  // Client is defined
  try{
    client.login();
  }catch(error){
    // handle error
    console.error("Got RPC error", error);
  }
}
```

it will of course work, but I will only catch the RPC timeout error. This is bad, because the RPC no socket error actually was fired immediately by `client.transport.connect()` in your code, but I cannot get it because that promise got rejected and it is unhandled by your code. You didn't `try...catch` it, and you didn't throw the value out so that I could handle it in my code. This, in turn, produces an `UnhandledPromiseRejectionWarning` and it makes me wait for the RPC timeout error to do anything useful, like inform the user to open Discord.

**Comment by xhayper** [comment]
Ahhh, i just need to pass the error from `client.transport.connect()` to the developer, i will see what i can do.

**Comment by xhayper** [comment]
Okay, i see the problem, i will fix it so the exception bubble up to the surface

**Comment by xhayper** [comment]
ok, this is weird, i am not sure why it doesn't bubble up at all

**Comment by xhayper** [comment]
I got it fixed

**Comment by xhayper** [comment]
![image](https://github.com/xhayper/discord-rpc/assets/46065349/1045c39e-5525-4273-b040-cb96eaf125c1)

**Comment by xhayper** [comment]
Fixed in 8634d96d2eddca171aa70251fec06eb946769b04
Publishing a new version

**Comment by xhayper** [comment]
Fixed in v1.0.17.
